java-rest-service/README.md

```diff
@@ -0,0 +1,220 @@
+# Introduction java-rest-service
+
+The java-rest-service app provides you an out-of-the-box application setup to implement your business logic.
+It is based on the commonly known 3-layered application architecture in where the package `api` provides the
+presentation layer, `domain` provides the services and business domain and finally the `data` package provides
+you the capability to persist your domain.
+
+It is leveraging Spring Boot as a technology stack, which provides:
+- a way to implement REST(ful) API using Spring Web annotations
+- generation of the OpenAPI definition based on your code
+- data persistence using Spring Data JPA (now PostgreSQL and MySQL is supported, but other databases can be easily added)
+- an Inversion of Control Container to wire together your classes at running without the need to write tightly-coupled code
+
+The application contains example code implementing REST API to write and read customer profile information to and from 
+database. This example is intended to showcase best practices around using Spring Boot and it's libraries as well as
+different types of tests which can be utilized to verify different parts of an application.
+
+## Prerequisites
+
+In order to further develop this application the following tools needs to be setup:
+- Java Development Kit
+- Visual Studio Code or IntelliJ IDEA as Integrated Development Environment (IDE)
+- Docker Desktop to execute integration tests or run the application locally
+
+## Local
+
+### Build
+
+In order to compile the production code:
+
+<!--- #IF(#buildTool == 'maven') -->
```

Comment:
Not a big deal, but why the three '-' (instead of 2)?

Reply:
oops :)